### DISCUSSION TITLE
php packages

### DISCUSSION THREAD

**Discussion by MarkusLange** [Novik/ruTorrent#2567] [Q&A]
Hello,

is there a list of needed php packages to run rutorrent proper?

**Comment by koblack** [comment]
I don't know of such a list, but from my experience Ubuntu 20.04 Focal Fossa needs
```
php-common
php-geoip
php-json
php-sqlite3
php-xml
php-zip
```
you can try to check it using

```
https://github.com/RogerGee/php-ext-depends/raw/master/depends.php
```
```
user@box:~$ php depends.php /var/www/html/ruTorrent/
Core (builtin)
ctype
date (builtin)
dom
fileinfo
geoip
iconv
json
libxml
pcre (builtin)
posix
session
SimpleXML
sockets
SPL (builtin)
sqlite3
standard (builtin)
sysvsem
xml
zip
zlib
```

**Comment by MarkusLange** [comment]
@TrimmingFool or @stickz can you confirm this list?

**Reply by stickz** [reply]
This information it outdated. The dependency check needs to be run on Ubuntu 22.04 LTS. 

`php-geoip` is gone. We are also missing certain things like `php-dom` and `php-mbstring`.

**Reply by MarkusLange** [reply]
yeah I have allready blocked the geoip plugin, php-mbstring is part of php8.2 (debian Bookworm) php-dem should be part of php-xml or not?

**Comment by MarkusLange** [comment]
On Ubuntu 22.04 LTS (arm64) with V4.2.6 I get this

ubuntu@ubuntu:~$ php depends.php /var/www/ruTorrent-4.2.6/
Core (builtin)
ctype
curl
date (builtin)
fileinfo
iconv
json (builtin)
libxml
mbstring
pcre (builtin)
posix
session
SPL (builtin)
standard (builtin)
sysvsem
zlib

Plugins geoip and rpc deactivated

ubuntu@ubuntu:~$ php -v
PHP 8.1.2-1ubuntu2.14 (cli) (built: Aug 18 2023 11:41:11) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.2, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.2-1ubuntu2.14, Copyright (c), by Zend Technologies

**Comment by MarkusLange** [comment]
@koblack thank you for the hint with the depends.php very nice tool
